[syamfarh] iP

syamfarh opened the pull request:
# Duke 
> "Try not. Do, or do not. There is no try. " — Yoda

Duke lessens the burden of recalling tasks, it is a 
- a text-based 
- user-friendly 
- lightning-fast program

You only need to 
1. download the jar file from [here](https://github.com/syamfarh/ip/releases/tag/A-Jar)
2. Copy the jar file into an empty folder
3. Open a command window in that folder
4. Run the command java -jar duke.jar

And it is **FREE!**  :heart_eyes:

Features available:
- [x] Task management
- [ ] Deadline management
- [ ] Reminders

If you're a Java programmer, you can utilize it for Java practice as well. Below is the main method:
```ruby
    public static void main(String[] args) {

        Duke chatBot = new Duke();
        chatBot.run();
        chatBot.exit();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
First commit
add name, greeting and exit

Commit:
Added Echo function to Duke

Commit:
Added ability to store user input

Commit:
added Task class.
added ability to mark task as done/undone.

Commit:
added ToDos, Events, Deadlines

Commit:
added automated text ui Testing.

Commit:
added error handling for
todo, deadline, event, invalid input

Commit:
added delete function from list.

Commit:
Add method to read task from file

Commit:
Add method to save tasks to file

Commit:
Add JavaDoc to fileFormat method

Commit:
Merge branch 'branch-Level-7'

Commit:
Add Date and Time Recognition.

currently only recognize dateTime format: dd/MM/yyyy HHmm

Commit:
Merge branch 'branch-Level-8'

Commit:
Abstract file processing into Storage class

Commit:
Abstract Arraylist Tasks into a TaskList class

Commit:
Abstract parsing of userInput to Parse class

Commit:
Abstract Ui from parser class

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Refactor classes into different package

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Commit:
Add Gradle support

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Add JUnit testing

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Fix outputList method not outputting DIVIDER lines

Commit:
Add javaDoc to methods in duke package

Commit:
Fix coding standards to all classes

Commit:
Add find function

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
fix merge Conflict

Commit:
Merge branch 'branch-Level-9'

matochichap commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+     * @return true if the str only contain numeric characters.
+     */
+    public static boolean isNumeric(String str) {
+        return str.matches("-?\\d+(\\.\\d+)?"); //match a number with optional '-' and decimal.
```
I love the use of the comment here!

matochichap commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    /**
+     * @return status icon of the task whether it is done or not.
```
Perhaps this could have a descriptor of the function

matochichap commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Read task from duke.txt file and copy to ArrayList list.
+     */
+    public ArrayList<Task> load() throws DukeException {
```
Perhaps the method name could tell what it is loading?

matochichap commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Write the task list to file.
+     */
+    public void save(ArrayList<Task> tasks) throws DukeException {
```
Perhaps the method name could tell what it is saving?

matochichap commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+        case "list":
+            ui.outputList(tasks);
+            break;
+        default:
```
maybe the contents of this could make use of some abstraction and separate if statements to avoid deep nesting?

matochichap commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+        }
+    }
+
+    public TaskList findTasks(String desc) {
```
perhaps desc could be spelt out in full so that there is no uncertainty of its meaning?

matochichap reviewed (commented):
Very readable code apart from the deeply nested switch statement in Parser. I love the very descriptive documentation which makes it easy to understand what is happening!

jingting1412 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+            ui.outputList(tasks);
+            break;
+        default:
+            if (input.startsWith("mark ")) {
```
Maybe can implement a Command class to store all the commands and their response?

jingting1412 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+     * @param tasks current Tasklist from chatBot.
+     * @param ui ui that control output result to user.
+     */
+    public static void replyUser(String input, TaskList tasks, Ui ui) throws DukeException {
```
Good job on the function names throughout! Clear and easy to understand 👍

jingting1412 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+        default:
+            if (input.startsWith("mark ")) {
+                try {
+                    int i = Integer.parseInt(input.substring(5));
```
Maybe naming the variable 'index' or 'taskNo' would be clearer in communicating the purpose of this variable?

jingting1412 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    private void stringToList(ArrayList<Task> tasks, String[] task) {
+
+        switch (task[0]) {
+        case "T":
```
Follows Java coding standard, good 👍

jingting1412 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return formatted String
+     */
+    public String fileFormat() {
```
Maybe this function name can be clearer in expressing its function?

jingting1412 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,46 @@
+public class ParserTest {
+
+    @Test
+    public void isValidDate_correctInput_test() {
```
Maybe adding the outcome for the test in the name makes its purpose more clear? E.g. isValidDate_correctInput_returnTrue()

jingting1412 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Test
+    public void replyUser_invalidInput2_test() {
```
Similar to previous comment, maybe including the outcome like replyUser_invalidInput2_errMsg() is better naming convention?

jingting1412 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+public class Storage {
+
+    /**
+     * Read task from duke.txt file and copy to ArrayList list.
```
Maybe can improve javadoc by adding the error that it throws

jingting1412 reviewed (commented):
Java coding standard closely followed with just a few small areas of improvement. Code is clear and easy to read. Good job 👍

Commit:
Add GUI but with bugs

Commit:
Fixed GUI

Commit:
Fixed checkstyle

Commit:
Merge branch 'branch-Level-10'

Commit:
Include data file in commit

Commit:
Add assertions to Tasklist methods

Ensure that the parameter passed is correct.

Let's add assert to all functions that accept parameter
in Tasklist class.

syamfarh committed:
Merge pull request #1 from syamfarh/branch-A-Assertions

Branch a assertions

Commit:
Abstract Ui string format

There are code duplication in the string format.

Let's abstract out the similar code into another function
that will append the similar format that is called in
every method that use similar format.

syamfarh committed:
Merge pull request #2 from syamfarh/branch-A-CodeQuality

Abstract Ui string format

Commit:
Merge branch 'master' of https://github.com/syamfarh/ip

Commit:
Add Help command that guide user on how to input command

syamfarh committed:
Merge pull request #3 from syamfarh/branch-C-Help

Add Help command that guide user on how to input command

Commit:
Merge branch 'master' of https://github.com/syamfarh/ip

* 'master' of https://github.com/syamfarh/ip:
  Add Help command that guide user on how to input command

Commit:
Fix javadocs in Duke class and renamed boolean isExit

Commit:
Abstract commands from Parser Class

Commit:
Fix checkstyle and naming convention

Commit:
Add representative screenshot

Commit:
Update the readme.md file

Commit:
Fix bug where chatbot doesn't work without data file
